### PR TITLE
Add movement of pest based on number

### PR DESCRIPTION
**Pull request by wenzeslaus**
Pest can now leave a cell if there is too much of units/individuals at one location.

### CODE REVIEW

File: include/pops/simulation.hpp [hunk 1]
```diff
@@ -470,6 +470,112 @@ class Simulation
...
+     */
+    template<typename DispersalKernel>
+    void move_units(
+        IntegerRaster& susceptible,
+        IntegerRaster& infected,
+        const IntegerRaster& total_hosts,
+        std::vector<std::tuple<int, int>>& outside_dispersers,
+        DispersalKernel& dispersal_kernel,
+        const std::vector<std::vector<int>>& suitable_cells,
+        double too_many_units,
```
Comment (wenzeslaus) [inline include/pops/simulation.hpp:509]: overpopulation_percentage

File: include/pops/simulation.hpp [hunk 2]
```diff
@@ -470,6 +470,112 @@ class Simulation
...
+        double leaving_units_ratio)
```
Comment (wenzeslaus) [inline include/pops/simulation.hpp:510]: leaving_ratio

File: include/pops/simulation.hpp [hunk 3]
```diff
@@ -470,6 +470,112 @@ class Simulation
...
+     * @param too_many_units
+     * @param leaving_units_ratio
+     *
+     * @note Exposed hosts are considered as not counting towards total number of units,
+     *       i.e., *total_host* is assumed to be S + E in SEI model.
+     * @note Mortality is not supported by this function, i.e., the mortality rasters
+     *       are not modified while the infected are.
+     */
+    template<typename DispersalKernel>
+    void move_units(
```
Comment (wenzeslaus) [inline include/pops/simulation.hpp:502]: move_pests

File: include/pops/simulation.hpp (hunk 2 above)
Comment (wenzeslaus) [reply include/pops/simulation.hpp]: leaving_percentage now

File: include/pops/simulation.hpp (hunk 3 above)
Comment (wenzeslaus) [reply include/pops/simulation.hpp]: move_overpopulated_pests

**Comment by wenzeslaus** [comment]
I now went with:

* `use_overpopulation_movements` (in config)
* `overpopulation_percentage` (in config)
* `leaving_percentage` (in config)
* `move_overpopulated_pests` (function in Simulation)
* `test_overpopulation_movements` (test)

**Comment by wenzeslaus** [comment]
@ChrisJones687 The code almost the same as what you have seen before. There is one more test. If you can check the terminology that would be great.

**Review by ChrisJones687 — APPROVED** [review]
Looks great! I made a few modifications to the documentation.

**Comment by wenzeslaus** [comment]
> I made a few modifications to the documentation.

Thanks. I forgot to search for remaining "units". Done now. Merging.

**Pull request merged**